### PR TITLE
fix(core): cap behavior-model scan sequence length before scan hang

### PR DESCRIPTION
**Pull request by rama0x1**
## Summary

`alberta_framework/core/behavior_model.py`'s `run_behavior_model_from_arrays`
hands its caller-supplied `observations`/`actions` step arrays straight to
`jax.lax.scan` with **no bound on the leading (step) axis**. A hostile or
mistaken caller supplying a huge array forces JAX to materialize per-step
outputs (and stacked inputs) at that length, hanging the process or
exhausting memory well before any step executes.

This is the same hang class already fixed for other scan-driven array loops
in this repo (`core/sarsa.py` #1996, `core/partner_policy_fusion.py` #1991,
`utils/nexting.py` #1992, `core/learners.py` #1989,
`core/average_reward.py` #1999) — `behavior_model.py`'s array runner was not
covered by any of those. `run_behavior_model_from_arrays` is exported public
API (`alberta_framework.__init__` and `alberta_framework.core.__init__`).

The existing per-call preflight in this file
(`_preflight_behavior_model_update_working_set`, from a prior fix) only
bounds the working set of a *single* `update`/`init` call by `n_actions` /
`feature_dim`; it says nothing about the `num_steps` leading axis that
`run_behavior_model_from_arrays` feeds to `jax.lax.scan`, so this is a
distinct gap.

## Fix

Adds `_require_behavior_model_sequence_length` (exact-type trusted-array +
shape gate, rejects a leading length outside `[1, 50_000]` before any scan)
and calls it at the top of `run_behavior_model_from_arrays`, before
`_integer_action_ids` and `jax.lax.scan` run. The `50_000` ceiling matches
the convention already established in `average_reward.py`
(`_AVERAGE_REWARD_SEQUENCE_MAX_STEPS`) and `learners.py`
(`_LEARNING_LOOP_MAX_STEPS`), with generous headroom above this module's
largest currently-exercised sequence (12 steps, in
`test_scan_loop_and_jit_compatibility`).

## Scope

- `alberta_framework/core/behavior_model.py`
- `tests/test_behavior_model.py`

## Verification

Commit under test: `8e75dbe7065e68ccb99a4d9609c10e7880c711cc`

```text
# On origin/main ec035f73 (pre-fix): no cap exists on
# run_behavior_model_from_arrays; jax.lax.scan is called unconditionally
# with the caller-supplied leading length. Confirmed by reading the
# pre-fix source (no _require_* length gate before jax.lax.scan).

# At this head: a monkeypatch spy on
# alberta_framework.core.behavior_model.jax.lax.scan proves it is never
# invoked for an oversized (50_001-step) or mismatched-length input;
# both are rejected with a ValueError before the scan runs. The
# 50_000-step boundary is exercised exactly (accepted at 50_000,
# rejected at 50_001), and the existing 12-step
# test_scan_loop_and_jit_compatibility / test_run_behavior_model_from_arrays_
# still_runs_inside_the_ceiling continue to pass unchanged.

.venv/bin/python -m pytest tests/test_behavior_model.py -q -o addopts=""
................................................................         [100%]
64 passed in 14.66s   # 55 pre-existing + 9 new

.venv/bin/python -m ruff check alberta_framework/core/behavior_model.py tests/test_behavior_model.py
All checks passed!

.venv/bin/python -m mypy
Success: no issues found in 197 source files

python -c "import alberta_framework; from alberta_framework import run_behavior_model_from_arrays"
import OK
```

- [x] logs: local pytest/ruff/mypy output above; 64 passed in tests/test_behavior_model.py (+9 new), full-project mypy strict clean (197 files), package import surface intact

- [x] domain-artifact: N/A - scan sequence-length hang guard, no IPMNIST/benchmark shard produced

## Deviations from pre-registration

None — this is a Fix (repairs a measurement-corrupting/hang-inducing gap),
not a benchmark climb, so no pre-registered comparison applies.

## Open / unexplained

None known. The `50_000` ceiling is a documented-protocol convention shared
with sibling fixes, not a hard resource-derived bound; if a legitimate
caller needs more steps, that ceiling should be raised deliberately alongside
new evidence of the larger workload, not silently.

- AI assistance: yes
- Model(s) used: anthropic/claude-sonnet-5
- Agent tooling: Claude Code
- Skill path: contribute-to-asi (local skill copy)
- Provenance status: self-reported

AI provider/model: Anthropic / claude-sonnet-5
Client / agent tooling: Claude Code
Attribution status: self-reported

---

### CODE REVIEW

**Comment by lalalune** [comment]
Superseded by merged PR #2060, which consolidates this resource boundary with shared scan/work budgets, combined-work checks, and the broader regression panel.
